### ISSUE TITLE
[FIX] Only 1 cast entry (actor) is shown

### ISSUE THREAD

**Issue by icore77** [gboudreau/XBMCnfoMoviesImporter.bundle#86]
hello,
I'm using your XBMC Local Media Scanner for movies and I encounter an issue with Casts list.
I generate a separated XML file with multiple xpath but only the first appears in movie details :
![2017-03-25_152422](https://cloud.githubusercontent.com/assets/26677743/24325288/c6cfb3b2-1195-11e7-80b5-1008366fd130.png)

I have tried in different cases and always only the first cast appears instead of all mentioned in XML file (each actor entry is specified with his name and thumb).
I have open the Database and it seems the issue come from the database is not filled correctly :
![2017-03-25_201339](https://cloud.githubusercontent.com/assets/26677743/24325367/ae98ccf0-1197-11e7-8fb0-ddf4be140700.png)

With default Plex Movie Scanner, all casts are showned correctly with thumbs.
With yours, only the first.
Can you fix it ?
How can I help you to troubleshoot this ?

Best regards.

**Comment by SlrG** [comment]
Please upload a zip of the movie folder, without the movie file(s) itself. It should contain the nfo file and all other metadata like poster, art etc.

**Comment by icore77** [comment]
Only 2 files (movie not included), XML and JPG.
If I invert User 1 and User 2, (User 2 became the first in order), User 2 appears but not User 1.
[Tears of Steel (2012).zip](https://github.com/gboudreau/XBMCnfoMoviesImporter.bundle/files/870794/Tears.of.Steel.2012.zip)

As mentioned, I think the problem comes from the data injection of multiple entries in database because the database content has only User 1 and not User 2.

Regards.

**Comment by SlrG** [comment]
The problem is, that you have no role tags for your actors. If that is missing, the agent assigned unknown to the role. Adding the second actor with the same "unknown" role is blocked by plex as two actors can't have the exact same role.  I changed the agent to add the number of the actor, too. So now it gets name Unknown Role x and thus is accepted by plex. Please update the agent to the newest version. Also please name the file moviename.nfo not moviename.xml. If it works for you, please close the issue.

**Comment by icore77** [comment]
Nice job !
When <role> xpath is not specified, Unknown Role x is specified, that's OK and works very well, thanks.
But another issue is if I set <role> xpath with same value "Himself" for example, the issue occurs too.
I understand this is a plex limitation and I purpose  you to add a "x" integer after the role (like Unknown Role) if a previous role as been found.

In attached file an example.
[Tears of Steel (2012).zip](https://github.com/gboudreau/XBMCnfoMoviesImporter.bundle/files/870870/Tears.of.Steel.2012.zip)

Regards.

**Comment by SlrG** [comment]
Makes sense. I will try to add this.

**Comment by SlrG** [comment]
@icore77 
Please give the newest version of the movies agent a try. I hope it does what you requested.

**Comment by icore77** [comment]
Excellent !
Just tried and works like  charm, and works with any value in "role" tag.
And if the "role" is found more than one time, an integer is added at the end. It's exactly what needed.

Very very nice job and a huge big thanks for your help.
Best regards.

**Comment by SlrG** [comment]
No problem. I'm glad it works and you like it. :)
